### ISSUE TITLE
[DOC] Script Request Processor - Non Primitive Types

### ISSUE THREAD

**Issue by macohen** [opensearch-project/documentation-website#4690]
**What do you want to do?**
Once dev is done on  https://github.com/opensearch-project/OpenSearch/issues/7757, we'll need to update documentation for how to use non-primitive types in the script request processor. 

- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

**Comment by macohen** [comment]
We're not going to release this in 2.11. We've had to pick up other work instead and have this on hold for now.

**Comment by hdhalter** [comment]
@Hi @macohen , is there any chance of this going out in 2.12? Thanks.

**Comment by hdhalter** [comment]
Related issue is closed so closing this PR.
